### PR TITLE
Integration - Errors

### PR DESCRIPTION
**Pull request by AleMathus**
### **Summary**
Added a progress bar to show the user that the request is being processed, and also added handling to errors such as having no internet connection.

### **Trello**

https://trello.com/c/FuYyt79o

### **Screenshot**

https://user-images.githubusercontent.com/108141974/182475752-0798d3f3-f5db-4af3-8630-a7ba6988d6b7.mov

### CODE REVIEW

File: app/src/main/java/com/example/myapplication/vm/LoginViewModel.kt [hunk 1]
```diff
@@ -50,22 +57,53 @@ class LoginViewModel(app: Application) : AndroidViewModel(app) {
...
+        }
+    }
+
+    private fun isOnline(context: Context): Boolean {
+        val connectivityManager =
+            context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        val capabilities =
+            connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)
+        if (capabilities != null) {
+            if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)) {
```
Comment (LisandroDiMeo) [inline app/src/main/java/com/example/myapplication/vm/LoginViewModel.kt]: You can have an (OR) here:
return capabilities.has(A) || capabilities.has(B) || capabilities.has(C).

**Review by LisandroDiMeo — APPROVED** [review]
Looks great! Just change that if branch I mentioned and we are ready to go!

File: app/src/main/java/com/example/myapplication/ui/LoginFragment.kt [hunk 2]
```diff
@@ -75,16 +77,38 @@ class LoginFragment : Fragment() {
...
+                        context,
+                        INTERNET,
+                        Toast.LENGTH_SHORT
+                    )
+                        .show()
+                }
+                NetworkResponseState.EXCEPTION -> {
+                    Toast.makeText(
+                        context,
+                        INTERNET,
```
Comment (ffasitella) [inline app/src/main/java/com/example/myapplication/ui/LoginFragment.kt]: Just a silly comment. This is a generic error, so we should show a generic message. What about "Something has failed, try again later"? 
Also, you could use the `else` branch here.

**Review by ffasitella — CHANGES_REQUESTED** [review]
Just a single comment! Nice work Ale!

**Review by jcruz-wolox — APPROVED** [review]

**Review by ffasitella — APPROVED** [review]
🚀

**Pull request merged**